msu/utilities/misc.nut

```diff
@@ -57,3 +57,102 @@
 
 	return ::isKindOf(obj, _className);
 }
+
+::MSU.serializeObject <- function( _object, _out )
+{
+	::MSU.requireOneOfType(["array", "table"], _object);
+	
+	local type = typeof _object;
+	local isTable = type == "table";
+
+	_out.writeString(type);
+	_out.writeU32(_object.len());
+
+	foreach (key, element in _object)
+	{
+		if (isTable) _out.writeString(key);
+		local elementType = typeof element;
+		_out.writeString(elementType);
+
+		switch (elementType)
+		{
+			case "integer":
+				if (element < 0)
+				{
+					_out.writeString("I");
+					_out.writeI32(element);
+				}
+				else
+				{
+					_out.writeString("U");
+					_out.writeU32(element);
+				}
+				break;
+
+			case "float":
+				_out.writeF32(element);
+
+			case "string":
+				_out.writeString(element);
+				break;
+
+			case "boolean":
+				_out.writeBool(element);
+				break;
+
+			case "array":
+			case "table":
+				::MSU.serializeObject(element, _out);
+				break;
+
+			default:
+				throw ::MSU.Exception.InvalidType(element);
```

Comment:
I'd wanna look into dealing with BB classes cuz in principle I don't see why we shouldn't be able to handle them, but I guess that can be done for 1.1.0

Reply:
So should we limit these functions to serialization of value types only for 1.0.0?

Reply:
I think it's fine to say we serialize standard tables, arrays and value types for now and then we can see where we go from there. I think this system works fine.